lib/job/arSearchResultExportCsvJob.class.php

```diff
@@ -0,0 +1,235 @@
+    if (null !== $criterias = $this->parseQuery())
+    {
+      $this->search->queryBool->addMust($criterias);
+    }
+
+    // Add criteria fo secondary search fields
+    foreach (SearchAdvancedAction::$NAMES as $name)
+    {
+      if (
+        !empty($this->searchParams[$name])
+        && (null !== $criterias = SearchAdvancedAction::fieldCriteria($name, $this->searchParams[$name]))
+      )
+      {
+        $this->search->queryBool->addMust($criterias);
+      }
+    }
+
+    // Set query if criteria were added
+    if (count($this->search->queryBool->getParams()))
+    {
+      $this->search->query->setQuery($this->search->queryBool);
+    }
+  }
+
+  /**
+   * Translate array of search parameters to query criteria.
+   *
+   * Modified version of parseQuery method in the SearchAdvancedAction class
+   *
+   * Each set of parameters is numbered, starting at zero, and includes three
+   * properties: query text (prefixed by "sq"), operation (prefixed by "so": "and" or
+   * "or"), and fields (prefixed by "sf") to return (defaulting to "_all").
+   *
+   * For example:
+   *
+   *   $this->searchParams = array(
+   *     'so0' => 'and',
+   *     'sq0' => 'cats',
+   *     'sf0' => ''
+   *   );
+   *
+   * @return object  \Elastica\Query\Bool instance
+   */
+  protected function parseQuery()
+  {
+    $queryBool = new \Elastica\Query\Bool();
+
+    $count = -1;
+
+    while (null !== $query = $this->searchParams['sq'.++$count])
+    {
+      if (empty($query)) continue;
+
+      $field = $this->searchParams['sf'.$count];
+      if (empty($field))
+      {
+        $field = '_all';
+      }
+
+      $operator = $this->searchParams['so'.$count];
+      if (empty($operator))
+      {
+        $operator = 'or';
+      }
+
+      $queryField = SearchAdvancedAction::queryField($field, $query, $this->archivalStandard);
+      SearchAdvancedAction::addToQueryBool($queryBool, $operator, $queryField);
+    }
+
+    if (0 == count($queryBool->getParams()))
+    {
+      return;
+    }
+
+    return $queryBool;
+  }
+
+  /**
+   * Export search results as CSV
+   *
+   * @param string  Path of file to write CSV data to
+   *
+   * @return int  Number of descriptions exported
+   */
+  protected function exportResults($path)
+  {
+    $itemsExported = 0;
+
+    // Exporter will create a new file each 10,000 rows
+    $writer = new csvInformationObjectExport(
+      $path,
+      $this->archivalStandard,
+      10000
+    );
+
+    // Force loading of information object configuration, then modify writer
+    // configuration
+    $writer->loadResourceSpecificConfiguration('QubitInformationObject');
+    array_unshift($writer->columnNames, 'referenceCode');
+    array_unshift($writer->standardColumns, 'referenceCode');
+
+    $resultSet = QubitSearch::getInstance()->index->getType('QubitInformationObject')->search($this->search->query);
+
+    foreach ($resultSet as $hit)
+    {
+      $resource = QubitInformationObject::getById($hit->getId());
```

Comment:
I know it's an edge case, but what if the ES index and MySQL get out of sync? We could get null returned by QubitInformationObject::getById() if our ES index thinks an info object is there when it isn't (I actually get this often when developing, when dealing with sql dumps and the like, but forgetting to search:populate again after). I just mention it since I ran into this when testing this feature and exportResource() crashes if you pass it null.

Maybe we could add here:
```
if (!$resource)
{
  $this->info('ES search result information object not found in MySQL, skipping... (you may need to re-run search:populate)');
  continue;
}
```